configlib/config.go

```diff
@@ -38,9 +38,61 @@ func NewConfig(cfg *PkgrConfig) {
 		rVersion := rcmd.GetRVersion(&rs)
 		cfg.Library = getLibraryPath(cfg.Lockfile.Type, cfg.RPath, rVersion, rs.Platform, cfg.Library)
 	}
+
+	// For all cfg values that can be paths, make sure that ~ is expanded to the home directory.
```

Comment:
Nice implementation. Just basically inject the changes before they would be used.